src/sentry/runner/commands/cleanup.py

```diff
@@ -398,6 +290,214 @@ def is_filtered(model: type[Model]) -> bool:
+        ).delete()
+
+
+def remove_expired_values_for_org_members(
+    is_filtered: Callable[[type[Model]], bool], days: int
+) -> None:
+    from sentry.models.organizationmember import OrganizationMember
+
+    debug_output("Removing expired values for OrganizationMember")
+    if is_filtered(OrganizationMember):
+        debug_output(">> Skipping OrganizationMember")
+    else:
+        expired_threshold = timezone.now() - timedelta(days=days)
+        OrganizationMember.objects.delete_expired(expired_threshold)
+
+
+def delete_api_models(is_filtered: Callable[[type[Model]], bool]) -> None:
+    from sentry.models.apigrant import ApiGrant
+    from sentry.models.apitoken import ApiToken
+
+    for model_tp in (ApiGrant, ApiToken):
+        debug_output(f"Removing expired values for {model_tp.__name__}")
+
+        if is_filtered(model_tp):
+            debug_output(f">> Skipping {model_tp.__name__}")
+        else:
+            queryset = model_tp.objects.filter(
+                expires_at__lt=(timezone.now() - timedelta(days=API_TOKEN_TTL_IN_DAYS))
+            )
+
+            # SentryAppInstallations are associated to ApiTokens. We're okay
+            # with these tokens sticking around so that the Integration can
+            # refresh them, but all other non-associated tokens should be
+            # deleted.
+            if model_tp is ApiToken:
+                queryset = queryset.filter(sentry_app_installation__isnull=True)
+
+            queryset.delete()
+
+
+def exported_data(is_filtered: Callable[[type[Model]], bool], silent: bool) -> None:
+    from sentry.data_export.models import ExportedData
+
+    if not silent:
+        click.echo("Removing expired files associated with ExportedData")
+
+    if is_filtered(ExportedData):
+        debug_output(">> Skipping ExportedData files")
+    else:
+        export_data_queryset = ExportedData.objects.filter(date_expired__lt=(timezone.now()))
+        for item in export_data_queryset:
+            item.delete_file()
+
+
+def models_which_use_deletions_code_path() -> list[tuple[type[Model], str, str]]:
+    from sentry.models.artifactbundle import ArtifactBundle
+    from sentry.models.eventattachment import EventAttachment
+    from sentry.models.grouprulestatus import GroupRuleStatus
+    from sentry.models.rulefirehistory import RuleFireHistory
+    from sentry.monitors.models import MonitorCheckIn
+    from sentry.replays.models import ReplayRecordingSegment
+
+    # Deletions that use the `deletions` code path (which handles their child relations)
+    # (model, datetime_field, order_by)
+    return [
+        (EventAttachment, "date_added", "date_added"),
+        (ReplayRecordingSegment, "date_added", "date_added"),
+        (ArtifactBundle, "date_added", "date_added"),
+        (MonitorCheckIn, "date_added", "date_added"),
+        (GroupRuleStatus, "date_added", "date_added"),
+        (RuleFireHistory, "date_added", "date_added"),
```

Comment:
I've made sure to add this as well:
https://github.com/getsentry/sentry/pull/77289/files#diff-af1cde82f065157a87af5b9c019c93a9e93aeb22661494449dbca1628e3cf306R223